### PR TITLE
Improve imbalance fee handling for edge cases

### PR DESCRIPTION
**Pull request by karlb**
## Description

Improve imbalance fee handling for edge cases
    
I'm pushing this early to avoid conflicts with @palango's work.
    
Follow-up todos:
* add one more edge case
* make tests more concise

Relates to https://github.com/raiden-network/raiden/issues/4858

## PR review check list

Quality check list that cannot be automatically verified.

- Safety
    - [ ] The changes respect the necessary conditions for safety (https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#protocol-values-constraints)
-  Code quality
    - [ ] Error conditions are handled
    - [ ] Exceptions are propagated to the correct parent greenlet
    - [ ] Exceptions are correctly classified as recoverable or unrecoverable
- Compatibility
    - [ ] State changes are forward compatible
    - [ ] Transport messages are backwards and forward compatible
- Commits
    - [ ] Have good messages
    - [ ] Squashed unecessary commits
- If it's a bug fix:
    - Regression test for the bug
        - [ ] Properly covers the bug
        - [ ] If an integration test is used, it could not be written as a unit test
- Documentation
    - [ ] A new CLI flag was introduced, is there documentation that explains usage?
- Specs
    - [ ] If this is a protocol change, are the specs updated accordingly? If so, please link PR from the specs repo.
- Is it a user facing feature/bug fix?
    - [ ] Changelog entry has been added

### CODE REVIEW

File: raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py [hunk 1]
```diff
@@ -1966,6 +1967,82 @@ def test_imbalance_penalty_at_insufficent_transfer_balance():
...
+    Test that having an imbalance penalty fee during a transfer where mediator has
+    insufficient balance does not throw an UndefinedMediationFee exception from
+    the state machine.
+    """
+    payer_transfer = create(
+        LockedTransferSignedStateProperties(amount=10, initiator=HOP1, target=ADDR, expiration=50)
+    )
+
+    imbalance_penalty = calculate_imbalance_fees(
+        channel_capacity=20, proportional_imbalance_fee=4000
```
Comment (palango) [inline raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py]: Is that even a fee of one token wei? 20*4000/1e6=0,08

Comment (karlb) [reply raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py]: For this test, it only matters that we try to calculate the IP and check is that succeeds. The result does not actually matter.

File: raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py [hunk 2]
```diff
@@ -1966,6 +1967,82 @@ def test_imbalance_penalty_at_insufficent_transfer_balance():
...
+        channelidentifiers_to_channels=channels.channel_map,
+        pseudo_random_generator=random.Random(),
+        block_number=2,
+    )
+    assert not pair
+
+
+def test_imbalance_penalty_with_barely_sufficient_balance():
+    """
+    Without the keeping the flat fee, the mediator's balance would be
```
[the remark this answers is not in the document]
Comment (karlb) [reply raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py]: It is. Fixed.

File: raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py (hunk 1 above)
Comment (palango) [reply raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py]: Ok, can you add a comment about that?

Comment (karlb) [reply raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py]: I've added such a comment to all three tests where this is the case. I'll try to clean up the tests later, so that there is less code and redundancy.

Comment (palango) [reply raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py]: Thanks!
